sale_order_type/models/stock_picking.py

```diff
@@ -13,7 +13,10 @@ class StockPicking(models.Model):
     def _create_invoice_from_picking(self, picking, vals):
         if picking and picking.sale_id:
             sale = picking.sale_id
-            if sale.type_id and sale.type_id.journal_id:
-                vals['journal_id'] = sale.type_id.journal_id.id
+            if sale.type_id:
```

Comment:
For not dropping so much the coverage you can do:
```
            vals['journal_id'] = (sale.type_id.journal_id.id if
                                  vals['type'] == 'out_invoice' else
                                  sale.type_id.refund_journal_id.id)
```
It's safe to do it (you don't need to check for existing values. If for example `sale.type_id.journal_id` is an empty recordset, then `sale.type_id.journal_id.id` will return `False`, not an error.